examples/vits.py

```diff
@@ -0,0 +1,624 @@
+    if not cleaner: raise ModuleNotFoundError('Unknown cleaner: %s' % name)
+    text = cleaner(text)
+  return [_symbol_to_id[symbol] for symbol in text]
+def expand_abbreviations(text):
+  for regex, replacement in _abbreviations: text = re.sub(regex, replacement, text)
+  return text
+def lowercase(text): return text.lower()
+def collapse_whitespace(text): return re.sub(_whitespace_re, ' ', text)
+def convert_to_ascii(text): return unidecode(text)
+def basic_cleaners(text): return collapse_whitespace(lowercase(text))
+def transliteration_cleaners(text): return collapse_whitespace(lowercase(convert_to_ascii(text)))
+def english_cleaners(text): return collapse_whitespace(phonemize(expand_abbreviations(lowercase(convert_to_ascii(text))), language='en-us', backend='espeak', strip=True))
+def english_cleaners2(text): return collapse_whitespace(phonemize(expand_abbreviations(lowercase(convert_to_ascii(text))), language='en-us', backend='espeak', strip=True, preserve_punctuation=True, with_stress=True))
+def intersperse(lst, item):
+  result = [item] * (len(lst) * 2 + 1)
+  result[1::2] = lst
+  return result
+def get_text(text, hps):
+  text_norm = text_to_sequence(text, hps.data.text_cleaners)
+  if hps.data.add_blank: text_norm = intersperse(text_norm, 0)
+  return Tensor(text_norm, dtype=dtypes.int64)
+VITS_PATH = Path(__file__).parent.parent / "weights/VITS/"
+MODELS = { # config_path, weights_path, config_url, weights_url
+  "ljs": (VITS_PATH / "ljs_base.json", VITS_PATH / "pretrained_ljs.pth", "https://raw.githubusercontent.com/jaywalnut310/vits/main/configs/ljs_base.json", "https://drive.google.com/uc?export=download&id=1q86w74Ygw2hNzYP9cWkeClGT5X25PvBT&confirm=t"),
+  "vctk": (VITS_PATH / "vctk_base.json", VITS_PATH / "pretrained_vctk.pth", "https://raw.githubusercontent.com/jaywalnut310/vits/main/configs/vctk_base.json", "https://drive.google.com/uc?export=download&id=11aHOlhnxzjpdWDpsz1vFDCzbeEfoIxru&confirm=t")
+}
+MODEL_TO_USE = "vctk" # "ljs" (female) or "vctk" (male)
+OUT_PATH = Path(__file__).parent.parent / f"temp/tts_vits_{MODEL_TO_USE}_test.wav"
+TEXT_TO_SYNTHESIZE = "Hello person, it's nice to talk with you, 1, 2, 3, weeee."
+# PAPER: https://arxiv.org/abs/2106.06103  CODE: https://github.com/jaywalnut310/vits/tree/main
+if __name__ == '__main__':
+  logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+  logger = logging
+  Tensor.no_grad, Tensor.Training = True, False
+  Tensor.manual_seed(1337)
+  np.random.seed(1337)
+  net_g, hps = load_model(MODELS[MODEL_TO_USE])
+  stn_tst = get_text(TEXT_TO_SYNTHESIZE, hps)
+  logger.info(f"Converted input text to tensor \"{TEXT_TO_SYNTHESIZE}\" -> Tensor({stn_tst.shape}): {stn_tst.numpy()}")
+  x_tst, x_tst_lengths = stn_tst.unsqueeze(0), Tensor([stn_tst.shape[0]], dtype=dtypes.int64)
+  start_time = time.time()
+  out = net_g.infer(x_tst, x_tst_lengths, sid=Tensor([4], dtype=dtypes.int64) if MODEL_TO_USE == "vctk" else None, noise_scale=.667, noise_scale_w=0.8, length_scale=1)
```

Comment:
If your counting the time it takes to infer, make sure you realize here so that it correctly measures all the time.

Reply:
AH oops, when I was refactoring stuff I forgot to change that, cheers!